include/server/api/base_parameters_grammar.hpp

```diff
@@ -149,10 +159,13 @@ struct BaseParametersGrammar : boost::spirit::qi::grammar<Iterator, Signature>
             qi::lit("bearings=") >
             (-(qi::short_ > ',' > qi::short_))[ph::bind(add_bearing, qi::_r1, qi::_1)] % ';';
 
+        sides_rule = qi::lit("sides=") >
```

Comment:
@fijemax it crashes because grammars are local variables in the constructor.

What i suggest is to use in the way as in commit https://github.com/oxidase/osrm-backend/commit/301a1cca6c5a6f3ee229213448577d9753e337d5 It contains more restrictive grammar [bod|eps]

Please add also some unit tests for parsing incorrect URLs as suggested by @TheMarex